app/javascript/src/LoginPage/loginForms/SignupForm.jsx

```diff
@@ -53,68 +67,68 @@ const InputFormGroup = (props) => {
 }
 
 class SignupForm extends Component<Props, State> {
-  constructor (props: Props) {
-    super(props)
-    this.state = {
-      username: this.props.user.username,
-      emailAddress: this.props.user.email,
-      firstname: this.props.user.firstname,
-      lastname: this.props.user.lastname,
-      password: '',
-      passwordConfirmation: '',
-      isFilledUsername: undefined,
-      isFilledEmailAddress: undefined,
-      isFilledFirstname: true,
-      isFilledLastname: true,
-      isFilledPassword: undefined,
-      isFilledPasswordConfirmation: undefined
-    }
+  state = {
+    username: this.props.user.username,
+    email: this.props.user.email,
+    firstName: this.props.user.firstname,
+    lastName: this.props.user.lastname,
+    password: '',
+    passwordConfirmation: '',
+    validation: {}
   }
 
-  handleTextInputUsername = (username: string) =>
-    this.setState({ username, isFilledUsername: username !== '' })
-
-  handleTextInputEmail = (emailAddress: string) =>
-    this.setState({ emailAddress, isFilledEmailAddress: emailAddress !== '' })
-
-  handleTextInputFirstname = (firstname: string) => this.setState({ firstname })
-
-  handleTextInputLastname = (lastname: string) => this.setState({ lastname })
+  handleInputChange = (value: string, event: SyntheticEvent<HTMLInputElement>) => {
+    const isValid = event.currentTarget.required ? validateSingleField(event) : true
+    const validation = {
+      ...this.state.validation,
+      [namesToStateKeys[event.currentTarget.name]]: isValid
+    }
 
-  handleTextInputPassword = (password: string) =>
     this.setState({
-      password,
-      isFilledPassword: password !== '',
-      isFilledPasswordConfirmation: this.state.passwordConfirmation === password
+      [namesToStateKeys[event.currentTarget.name]]: value,
+      validation
     })
+  }
 
-  handleTextInputPasswordConfirmation = (passwordConfirmation: string) =>
-    this.setState({
-      passwordConfirmation,
-      isFilledPasswordConfirmation: passwordConfirmation !== '' && passwordConfirmation === this.state.password
-    })
+  renameValidation = (namesToStateKeys: NamesToKeys, invalidFields: Validation) => Object.keys(invalidFields)
```

Comment:
Have you checked if validate.js can be assigned another format to avoid having this mapping function?

Reply:
Yes, I checked and it works with the `name` attribute: 
"This function will find all named inputs (inputs that specify the name attribute) and collect their values."
https://validatejs.org/#utilities-collect-form-values

The issue is for this page the name attribute comes in this format `user[username]`.  In the other forms is just `username`. And I don't want to add this names to the state `user[username]`, because it looks like object property accessors syntax